### PR TITLE
Inline Chat: Use intent detection for user input and support parallel chats

### PR DESCRIPTION
**Pull request by umpox**
## Description

This PR:
- Removes old `fixup` logic from the inline controller. All fixups now use the non-stop logic, we no longer need this.
- Updates the inline controller to support parallel inline chats. We no longer have a hard restriction on Cody only responding to one inline chat at a time.
- Adds LLM intent detection for the inline chat input so users do not need to use the /fix command to access fixups from inline chat. We are limited in what the text area can do, we can't show a helpful UI for commands like we do in the sidebar. Instead, we can leave this as purely focused on inline chat and inline fix. Users can still use `/fix` and `/chat` commands to bypass the LLM intent detection.
- Removes the `/touch` command. This is currently broken and is likely better handled for specific cases like generating tests. (Confirmed this doesn't block @abeatrix's work)\

https://github.com/sourcegraph/cody/assets/9516420/2b243732-afb4-47aa-9dea-292605d71a67

## Test plan

Locally tested:
- Inline chat and fix with LLM intent
- Inline chat and fix with specific commands
- Multiple inline chats
- Multiple inline chats + fixups

### CODE REVIEW

**Comment by umpox** [comment]
TODO: Figure out what to do if we have existing messages, maybe keep the chat around instead of deleting it, so users don't accidentally lose their chats

File: vscode/src/editor/vscode-editor.ts [hunk 1]
```diff
@@ -66,51 +66,12 @@ export class VSCodeEditor implements Editor<InlineController, FixupController, C
...
-        const activeEditor = vscode.window.visibleTextEditors.find(
-            editor => editor.document.uri.fsPath === inlineChatEditor.filePath
-        )
-        const selectionRange = this.controllers.inline?.getSelectionRange()
-        if (!activeEditor || !selectionRange) {
-            return null
-        }
-        const selection = new vscode.Selection(selectionRange.start.line, 0, selectionRange.end.line + 1, 0)
-        return this.createActiveTextEditorSelection(activeEditor, selection)
-    }
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply vscode/src/editor/vscode-editor.ts:102]: This means users can no longer run default and custom commands with inline chat...?

File: lib/shared/src/chat/recipes/inline-touch.ts [hunk 2]
```diff
@@ -71,7 +69,6 @@ export class InlineTouch implements Recipe {
             'selection',
             new BufferedBotResponseSubscriber(async content => {
                 if (!content) {
-                    await context.editor.controllers?.inline?.error()
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply lib/shared/src/chat/recipes/inline-touch.ts:74]: I saw in a recent issue where a user mentioned they use the touch command, so I can work on moving this to the custom prompt recipe before we remove this?

**Comment by umpox** [comment]
Closing as inline chat is being removed

**Pull request closed without merging**